.github/workflows/ci.yml

```
@@ -221,131 +175,139 @@ jobs:
+      - name: cargo clippy druid-shell
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --manifest-path=druid-shell/Cargo.toml --all-targets -- -D warnings
+        continue-on-error: true
+
+      - name: cargo clippy druid
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --manifest-path=druid/Cargo.toml --all-targets --features=svg,image,im -- -D warnings
+        continue-on-error: true
+
       # Test packages in deeper-to-higher dependency order
       - name: cargo test druid-shell
         uses: actions-rs/cargo@v1
         with:
           command: test
           args: --manifest-path=druid-shell/Cargo.toml --no-default-features --features=x11
+        continue-on-error: true
 
+      # We use --all-targets to skip doc tests, which are run in doctest-stable
       - name: cargo test druid
         uses: actions-rs/cargo@v1
         with:
           command: test
-          args: --manifest-path=druid/Cargo.toml --no-default-features --features=svg,image,im,x11
+          args: --manifest-path=druid/Cargo.toml --all-targets --no-default-features --features=svg,image,im,x11
+        continue-on-error: true
 
       - name: cargo test druid-derive
         uses: actions-rs/cargo@v1
         with:
           command: test
           args: --manifest-path=druid-derive/Cargo.toml
-        if: contains(matrix.os, 'windows')
+        continue-on-error: true
 
       - name: cargo test book examples
         uses: actions-rs/cargo@v1
         with:
           command: test
           args: --manifest-path=docs/book_examples/Cargo.toml
-        if: contains(matrix.os, 'windows')
+        continue-on-error: true
 
-  # we test the gtk backend as a separate job because gtk install takes
-  # a long time.
-  test-nightly-gtk:
-    runs-on: ubuntu-latest
-    name: cargo test nightly (gtk)
+  doctest-beta:
+    runs-on: macOS-latest
+    name: druid doctests beta
     steps:
       - uses: actions/checkout@v2
 
-      - name: install libgtk-3-dev
-        run: |
-          sudo apt update
-          sudo apt install libgtk-3-dev
-
-      - name: install stable toolchain
+      - name: install beta toolchain
         uses: actions-rs/toolchain@v1
         with:
-          toolchain: nightly
+          toolchain: beta
           profile: minimal
           override: true
 
       - name: restore cache
         uses: Swatinem/rust-cache@v1
 
-      - name: cargo test druid-shell
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --manifest-path=druid-shell/Cargo.toml
-
-      - name: cargo test druid
+      - name: cargo test druid --doc
         uses: actions-rs/cargo@v1
         with:
           command: test
-          args: --manifest-path=druid/Cargo.toml --features=svg,image,im
+          args: --manifest-path=druid/Cargo.toml --doc --no-default-features --features=svg,image,im,x11
```

do not enable x11 on macos :)
```suggestion
          args: --manifest-path=druid/Cargo.toml --doc --no-default-features --features=svg,image,im
```